### PR TITLE
Add visual prompting documentation

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

This PR includes a documentation for visual prompting.
image
image
image

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [x] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: docs/source/guide/explanation/algorithms/visual_prompting/index.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
+Visual Prompting
+=================
+
+Visual prompting is a computer vision task that uses a combination of an image and prompts, such as texts, bounding boxes, points, and so on to troubleshoot problems.
+Using these useful prompts, the main purpose of this task is to obtain labels from unlabeled datasets, and to use generated label information on particular domains or to develop a new model with the generated information.
+
+This section examines the solutions for visual prompting offered by the OpenVINO Training Extensions library.
+`Segment Anything (SAM) <https://arxiv.org/abs/2304.02643>`_, is one of the most famous visual prompting methods and this model will be used to adapt a new dataset domain.
+Because `SAM <https://arxiv.org/abs/2304.02643>`_ was trained by using web-scale dataset and has huge backbone network, fine-tuning the whole network is difficult and lots of resources are required.
+Therefore, in this section, we try to fine-tune only mask decoder only for several epochs to increase performance on the new dataset domain.
```
Comment (sungmanc) [inline docs/source/guide/explanation/algorithms/visual_prompting/index.rst:10]:
```suggestion
Therefore, in this section, training the whole network will not be handled
```

Comment (sungmanc) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:10]: I think your previous intention is to introduce the fine-tune the mask decoder, if so, how about introducing the CLI example?

**Review by jaegukhyun — APPROVED** [review]

File: docs/source/guide/explanation/algorithms/visual_prompting/index.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+For fine-tuning `SAM <https://arxiv.org/abs/2304.02643>`_, we use following algorithms components:
+
+.. _visual_prompting_finetuning_pipeline:
```
Comment (sungmanc) [inline docs/source/guide/explanation/algorithms/visual_prompting/index.rst:13]: How about adding information about the Mask Encoder and Decoder what we used?

Comment (sungchul1) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:13]: I'm not sure if it is needed, because this section is for introducing visual prompting implemented on otx, not introducing SAM.

Comment (sungmanc) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:13]: Do we used the same encoder, decoder with SAM..? If so, no problems at all

File: docs/source/guide/explanation/algorithms/visual_prompting/index.rst (hunk 1 above)
Comment (sungchul1) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:10]: The default setting is fine-tuning only the mask decoder and we cannot choose which modules will be fine-tuned in CLI.
It is possible in `config.yaml`, but introducing modifying `config.yaml` to fine-tune other modules is not in this section's scope.
So, I think we don't need to introduce CLI example.

File: docs/source/guide/explanation/algorithms/visual_prompting/index.rst (hunk 2 above)
Comment (sungchul1) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:13]: Right, using SAM as it is.

File: docs/source/guide/explanation/algorithms/visual_prompting/index.rst (hunk 1 above)
Comment (sungmanc) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:10]: Hmm, if so, I think  this "Therefore, in this section, training the whole network will not be handled" sentence also need to be deleted since we didn't guide the fine-tune in this section.

Comment (sungchul1) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:10]: Yes, and I described "fine-tuning the whole network is difficult and lots of resources are required" in the above line, so "fine-tune only mask decoder only" should be there.

Comment (sungmanc) [reply docs/source/guide/explanation/algorithms/visual_prompting/index.rst:10]: Then "Therefore, finetuning mask decoder is only supported." is a proper sentence that removes the "in this section", I think. I couldn't find any `try` in this document section.

**Review by sungmanc — APPROVED** [review]
